src/spikeinterface/qualitymetrics/tests/test_metrics_functions.py

```diff
@@ -47,6 +52,108 @@
 job_kwargs = dict(n_jobs=2, progress_bar=True, chunk_duration="1s")
 
 
+def _small_sorting_analyzer():
+    recording, sorting = generate_ground_truth_recording(
+        durations=[2.0],
+        num_units=4,
+        seed=1205,
+    )
+
+    sorting = sorting.select_units([3, 2, 0], ["#3", "#9", "#4"])
+
+    sorting_analyzer = create_sorting_analyzer(recording=recording, sorting=sorting, format="memory")
+
+    extensions_to_compute = {
+        "random_spikes": {"seed": 1205},
+        "noise_levels": {"seed": 1205},
+        "waveforms": {},
+        "templates": {},
+        "spike_amplitudes": {},
+        "spike_locations": {},
+        "principal_components": {},
+    }
+
+    sorting_analyzer.compute(extensions_to_compute)
+
+    return sorting_analyzer
+
+
+@pytest.fixture(scope="module")
+def small_sorting_analyzer():
+    return _small_sorting_analyzer()
+
+
+def test_unit_structure_in_output(small_sorting_analyzer):
+    for metric_name in get_quality_metric_list():
+        result = _misc_metric_name_to_func[metric_name](sorting_analyzer=small_sorting_analyzer)
+
+        if isinstance(result, dict):
+            assert list(result.keys()) == ["#3", "#9", "#4"]
+        else:
+            for one_result in result:
+                assert list(one_result.keys()) == ["#3", "#9", "#4"]
+
+    for metric_name in get_quality_metric_list():
+        result = _misc_metric_name_to_func[metric_name](sorting_analyzer=small_sorting_analyzer, unit_ids=["#9", "#3"])
+
+        if isinstance(result, dict):
+            assert list(result.keys()) == ["#9", "#3"]
+        else:
+            for one_result in result:
+                print(metric_name)
+                assert list(one_result.keys()) == ["#9", "#3"]
+
+
+def test_unit_id_order_independence(small_sorting_analyzer):
+    """
+    Takes two almost-identical sorting_analyzers, whose unit_ids are in different orders and have different labels,
+    and checks that their calculated quality metrics are independent of the ordering and labelling.
+    """
+
+    recording, sorting = generate_ground_truth_recording(
+        durations=[2.0],
+        num_units=4,
+        seed=1205,
+    )
+    sorting = sorting.select_units([0, 2, 3])
```

Comment:
Since the goal is to make a second analyzer with re-sorted units, why not:
```suggestion
    recording = small_sorting_analyzer.recording
    sorting = small_sorting_analyzer.sorting.select_units([0, 2, 3])
```

Reply:
Oh yes, much better!